game/src/contrib/systems/HealthSystem.java

```diff
@@ -34,22 +34,31 @@ public void execute() {
                 .map(this::applyDamage)
                 // Filter all dead entities
                 .filter(hsd -> hsd.hc.isDead())
-                .filter(
-                        hsd -> {
-                            if (!hsd.ac.hasAnimation(AdditionalAnimations.DIE)
-                                    || hsd.ac
-                                            .getAnimation(AdditionalAnimations.DIE)
-                                            .get()
-                                            .isLooping()) return true;
-                            if (!hsd.ac.isCurrentAnimation(AdditionalAnimations.DIE)) {
-                                hsd.ac.currentAnimation(AdditionalAnimations.DIE);
-                            }
-                            return hsd.ac.currentAnimation().isFinished();
-                        })
+                .filter(this::checkAnimationStatus)
                 // Remove all dead entities
                 .forEach(this::removeDeadEntities);
     }
 
+    private boolean checkAnimationStatus(HSData hsd) {
```

Comment:
der methodenname verrät mir noch nicht viel, was der rückgabewert bedeutet ...?

Reply:
die methode setzt sich aus lauter bedingungen zusammen, die einzeln abgefragt und zurückgeliefert werden. ohne die kommentare würde das recht unübersichtlich werden.

ich frage mich grad, was passieren würde, wenn man je kommentar den code nochmal in eine hilfsstruktur auslagert. muss ja keine eigene methode sein, ein `Predicate<>` würde vermutlich jeweils reichen. könnte man dann eine art boolsche verknüpfung draus machen der art `return hasDeathAnimation.test() || isLooping.test() || isFinished.test();`?

Reply:
hab das erst mal in filterByAnimation umbenannt, das schien mir aktuell am sinnigsten

Das mit den Predicates hab ich auch umgesetzt, ist das ca. was du dir vorgestellt hast?

Reply:
Ich finde das gar nicht so schlecht :) ... Kommentare am neuen Code, deshalb mache ich diese "Diskussion" mal zu.